build-tools/packages/build-cli/src/library/packageExports.ts

```diff
@@ -331,3 +331,57 @@ export function getTypesPathFromPackage(
 
 	return typesPath;
 }
+
+/**
+ * Read package "exports" to determine which of given "default" paths are present in map.
+ *
+ * @param packageJson - json content of package.json
+ * @param mapSrcQueryPathToOutKey - keys of map represent paths to match.
+ * @param emitDeclarationOnly - If true, "types" exports are considered.
+ * @param logger - optional Logger
+ * @returns Map with API tags or levels with export path data
+ */
+export function mapSourceExportsPath<TOutKey>(
```

Comment:
nit: I would call this `mapRuntimeExportsPath`. Source is the TypeScript files. This is finding JavaScript files.

Reply:
or `mapDefaultExportsPath`.

Reply:
My mistake this is inferring a .ts file, but we shouldn't do that in here. This should just look up the file referenced. The caller can do the work to figure out the source. The source should actually come from the tsconfig.